index.js

```diff
@@ -115,24 +114,35 @@ function Client (opts) {
   this.config(opts)
   this._log = this._conf.logger || new NoopLogger()
 
-  // start stream in corked mode, uncork when cloud
-  // metadata is fetched and assigned.  Also, the
-  // _maybeUncork will not uncork until _encodedMetadata
-  // is set
-  this.cork()
-  this._fetchAndEncodeMetadata(() => {
-    // _fetchAndEncodeMetadata will have set/memoized the encoded
-    // metadata to the _encodedMetadata property.
-
-    // This reverses the cork() call in the constructor above. "Maybe" uncork,
-    // in case the client has been destroyed before this callback is called.
-    this._maybeUncork()
-
-    // the `cloud-metadata` event allows listeners to know when the
-    // agent has finished fetching and encoding its metadata for the
-    // first time
-    this.emit('cloud-metadata', this._encodedMetadata)
-  })
+  if (this._conf.cloudMetadataFetcher && this._conf.expectExtraMetadata) {
+    throw new Error('it is an error to create a Client with both cloudMetadataFetcher and expectExtraMetadata')
```

Comment:
It looks like this will hard crash a service or application if it's misconfigured.  Are we OK with this?  My general understanding is we  (APM Agents generally) try to log a configuration but keep a customer's application or service.  Do we need a try/catch in the agent code now?

Reply:
It will intentionally throw if the Agent misuses this http client. I.e. if the Agent has a bug.  There isn't anything a customer's application code can do to impact that.

Reply:
I see -- so long as the agent never sends through this configuration the client won't throw, and we'll ensure that the agent doesn't do this. 👍